libminifi/include/utils/OptionalUtils.h

```diff
@@ -0,0 +1,44 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef LIBMINIFI_INCLUDE_UTILS_OPTIONALUTILS_H_
+#define LIBMINIFI_INCLUDE_UTILS_OPTIONALUTILS_H_
+
+#include <utility>
+
+#include <nonstd/optional.hpp>
+#include "utils/GeneralUtils.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace utils {
+
+template<typename T>
+nonstd::optional<typename remove_cvref<T>::type> optional_from_nullable(T&& obj) {
```

Comment:
I recommend introducing an alias to optional in our namespace and referring to that in our code. This would simplify later extension and migration.

Reply:
Added: 
```c++
template <typename T>
using optional = nonstd::optional<T>;
const nonstd::nullopt_t nullopt = nonstd::nullopt;
```
However, for this to work there is no good way to refer `nonstd::nullopt` from outside of the helper namespace. Any improvement idea on this?

Reply:
why not `using nostd::nullopt;` ?

Reply:
Because it is a value, not a type. There is the added value declaration, but it can only be used by either declaring it in the caller scope or by providing a fully qualified name.

Reply:
I am not aware of such restrictions on the using-declaration: https://godbolt.org/z/w7jbKM

Reply:
You can use using-declarations on variables.

Reply:
Updated.